### ISSUE TITLE
[SPIR-V] HLSL clip is treated as OpKill instead of OpDemoteToHelperInvocation in Vulkan 1.3 or -fspv-extension=SPV_EXT_demote_to_helper_invocation

### ISSUE THREAD

**Issue by danginsburg** [microsoft/DirectXShaderCompiler#5937]
**Description**
DXC supports generating OpDemoteToHelperInvocation for HLSL discard if using Vulkan 1.3 or -fspv-extension=SPV_EXT_demote_to_helper_invocation.  This is correct.  But if you use `clip(-1)` in your shader you will still get OpKill.  Is this just a bug?

**Steps to Reproduce**
https://shader-playground.timjones.io/6c6b28f4a3a9fe0365846aff9b7aa2c7

```
struct PSInput
{
	float4 color : COLOR;
};

float4 PSMain(PSInput input) : SV_TARGET
{
    // Incorrect: OpKill
    clip(input.color.a < 0 ? -1 : 1);
    // Correct: OpDemoteToHelperInvocation
    //if (input.color.a < 0 )
      //  discard;
	return input.color;
}
```

**Actual Behavior**
I think clip should produce the same output as discard?

**Environment**
- DXC version : Updated from trunk on 2023-06-16
- Host Operating System : N/A

**Comment by sudonatalie** [comment]
@keenuts Can you take a look at this?

**Comment by s-perron** [comment]
Yes, this is a bug, and should be fixed. We should never be generating OpKill.

**Comment by jeremy-lunarg** [comment]
@s-perron

I believe this issue can be closed. You fixed this in #5970, but I think you referenced the wrong issue when you submitted.
